config/src/main/java/com/typesafe/config/impl/ConfigNodeKeyValue.java

```diff
@@ -0,0 +1,48 @@
+package com.typesafe.config.impl;
+
+import com.typesafe.config.ConfigNode;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+public class ConfigNodeKeyValue implements ConfigNode{
+    final private ArrayList<ConfigNode> children;
+    private int configNodeValueIndex;
+    private ConfigNodeKey key;
+    private ConfigNodeValue value;
```

Comment:
Hmm, sounds good. I'm guessing that should be fine, since in most cases I think a key and a value will probably only be checked once, and a key-value node will probably not contain a large number of children.